src/main/java/org/jahia/services/usermanager/ldap/LDAPUserGroupProvider.java

```diff
@@ -384,6 +388,25 @@ public boolean verifyPassword(String userName, String userPassword) {
                 // that needs to be removed from the user DN for the lookup to succeed.
                 ctx.lookup(LdapUtils.newLdapName(userCacheEntry.getDn()));
                 logger.debug("Password verified for {} in {} ms", userName, System.currentTimeMillis() - startTime);
+           
+                try {
+                    JCRTemplate.getInstance().doExecuteWithSystemSession((JCRSessionWrapper session) -> {
+                        final String userRelativePath = ServicesRegistry.getInstance().getJahiaUserManagerService().getUserSplittingRule().getRelativePathForUsername(userName);
+                        final String providerKey = getKey();
+                        final String siteKey = getSiteKey();
+                        final StringBuilder memberPath = new StringBuilder();
+                        if (siteKey != null) {
+                            memberPath.append("/sites/").append(siteKey);
+                        }
+                        memberPath.append("/users/providers/").append(providerKey).append(userRelativePath);
+                        JahiaGroupManagerService.getInstance().flushMembershipCache(memberPath.toString(), session);
+                        ldapCacheManager.clearUserCacheEntryByName(providerKey, userName);
+                        return Boolean.TRUE;
+                    });
+                } catch (RepositoryException ex) {
```

Comment:
I would have catch any exception in here, can you validate with a PM that any issue with the cache should not prevent a valid user to connect?

Reply:
Code updated :)